loggo/loggo.py

```diff
@@ -360,13 +360,19 @@ def emit(handler_self, record: logging.LogRecord) -> None:
     def _params_to_dict(function: Callable, *args: Any, **kwargs: Any) -> Optional[Mapping]:
         """
         Turn args and kwargs into an OrderedDict of {param_name: value}.
-        Returns None if binding arguments to function signature fails.
+        Returns None if getting the signature, or binding arguments to
+        the signature's parameters fails.
         """
-        sig = inspect.signature(function)
+        try:
+            sig = inspect.signature(function)
+        except ValueError:
```

Comment:
Maybe we should handle `TypeError` as well

Reply:
isn't that redundant if the signature says `function: Callable` ? The error would mean that a user decorated a non-callable or we have a bug. If we have a bug, I prefer the exception to be raised. If user decorates a non-callable, IMO that's their problem and should fix it --> also prefer us to raise.

Reply:
Agreed, a `TypeError` should never be raised if its a callable.